ingester/src/persist/worker.rs

```diff
@@ -412,9 +412,12 @@ where
                     )
                     .await
                 {
-                    Ok(_) => ControlFlow::Break(Ok(())),
+                    Ok(_) => ControlFlow::Break(Ok(new_sort_key_colids)),
                     Err(CasFailure::QueryError(e)) => ControlFlow::Continue(e),
-                    Err(CasFailure::ValueMismatch(observed)) if observed == new_sort_key_str => {
+                    Err(CasFailure::ValueMismatch((observed_sort_key, observed_sort_key_ids)))
+                        if observed_sort_key == new_sort_key_str
+                            && observed_sort_key_ids == Some(new_sort_key_colids.clone()) =>
+                    {
```

Comment:
Lets make this a bit more robust - if the sort keys with column names is identical, the sort keys with IDs MUST be identical:

```suggestion
                    Err(CasFailure::ValueMismatch((observed_sort_key, observed_sort_key_ids)))
                        if observed_sort_key == new_sort_key_str =>
                    {
                        // Invariant: if the column name sort keys match, the sort
                        // key IDs must also match.
                        assert_eq!(&observed_sort_key_ids, Some(&new_sort_key_colids));
```

The way the code is currently wrote, if the catalog becomes corrupt, the ingester will use it. With this, at least we'll stop!

Reply:
Done